tox.ini

```
@@ -0,0 +1,92 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[tox]
+skipsdist=True
+skip_missing_interpreters = True
+envlist = lint-mock, lint, unit, static, coverage-report, coverage-xml
+
+[vars]
+src_path = {toxinidir}/src/
+tst_path = {toxinidir}/tests/
+;lib_path = {toxinidir}/lib/charms/operator_name_with_underscores
+all_path = {[vars]src_path} {[vars]tst_path}
+
+[testenv]
+setenv =
+  PYTHONPATH = {toxinidir}:{toxinidir}/lib:{[vars]src_path}
+  PYTHONBREAKPOINT=ipdb.set_trace
+  PY_COLORS=1
+passenv =
+  PYTHONPATH
+  CHARM_BUILD_DIR
+  MODEL_SETTINGS
+
+[testenv:format]
+allowlist_externals = tox
+description = Apply coding style standards to code
+deps =
+    ruff
+commands =
+    ruff format {[vars]all_path}
+    ruff check --fix {[vars]all_path}
+
+[testenv:lint]
+description = Check code against coding style standards
+setenv =
+  PYTHONPATH = {envdir}{:}{[vars]lib_path}
+deps =
+    mypy
+    ruff
+    pydantic[mypy]
+    types-PyYAML
+commands =
+    ruff format --check {[vars]all_path}
+    ruff check {[vars]all_path}
+    mypy {[vars]all_path}
+
+[testenv:unit]
+runner = uv-venv-lock-runner
+description = Run unit tests
+dependency_groups = unit
+commands =
+    uv run coverage run --source={[vars]src_path},{[vars]lib_path} \
+        -m pytest --ignore={[vars]tst_path}integration -vv \
+        --basetemp={envtmpdir} \
+        --tb native -s {posargs}
+    uv run coverage report --show-missing
+
+[testenv:coverage-report]
+description = Create test coverage report
+deps =
+    coverage[toml]
+commands =
+    coverage report
+
+[testenv:coverage-xml]
+description = Create test coverage XML report
+deps =
+    coverage[xml]
+commands =
+    coverage xml
+
+[testenv:static]
+description = Run static analysis tests
+deps =
+    bandit[toml]
+commands =
+    bandit -c {toxinidir}/pyproject.toml -r {[vars]all_path}
+
+[testenv:{integration,integration-k8s,integration-etcd,integration-ceph,integration-upgrade}]
```

This is overboard...

```suggestion
[testenv:integration]
```